Main.py

```diff
@@ -1,82 +1,102 @@
                     if any(toMutate):
-                        mutateORF = np.random.random([sum(toMutate),1]) * (data1_Cum[-1]) # pick mutation
+                        mutateORF = np.random.random([sum(toMutate), 1]) * (data1_Cum[-1]) # pick mutation
                         idxORF = np.searchsorted(data1_Cum, mutateORF, side='right') # peek mutation
                         mutation[toBirth[toMutate]] = idxORF  # store mutation
-                        #idxDivide = np.nonzero(toDivide)[0]
-                        #nr_mutations +=
+                        # idxDivide = np.nonzero(toDivide)[0]
+                        # nr_mutations +=
                         print 'time is ', t
                         print 'alive: ', nr_alive
-                        print data4[idxORF,iEnv]
-
+                        print data4[idxORF, iEnv]
                         simEnv['cell_cycle_time'][toBirth[toMutate]] = simEnv['cell_cycle_time'][idxDivide[toMutate]] + \
-                                                                       data4[idxORF,iEnv]*MEAN_CELL_CYCLE_TIME #(CELL_CYCLE_TIMES[simEnv['founder_id'][idxDivide[toMutate]]])
+                                                                       data4[idxORF, iEnv]*MEAN_CELL_CYCLE_TIME #(CELL_CYCLE_TIMES[simEnv['founder_id'][idxDivide[toMutate]]])
                         simEnv['next_divition'][toBirth[toMutate]] = t + simEnv['cell_cycle_time'][toBirth[toMutate]]
 
                 ## Population ages
                 simEnv['age'][0:nr_alive] += 0.1 # worry, might take up alotta time.
                 nr_alive += nr_Divide
-                #print 'alive: ', nr_alive
+                # print 'alive: ', nr_alive
 
                 # Check cycle exit condition
-                if nr_alive >= MAXIMUM_NR_AGENTS:
-                    stillInCycle = 0
-
-
-
-            # \todo: Sample for next cycle
+                if __name__ == '__main__':
+                    if __name__ == '__main__':
+                        if nr_alive >= MAXIMUM_NR_AGENTS-10:
+                            stillInCycle = 0
+        
+                # \todo: Save data for Growth curve
+                    # if iCycle == 0 or 24 or 49:
+                    # nr_alive at increments of 20 min. -> 30 points
+                    
+                # \todo: Save data for Evolutionary tree, Look plot code online. Binary tree!
+                    # mutations and mothers for all mutated cells
+                    # nr unmutated cells, 
+                    # time of mutation                        
+    
+                # \todo: Save data for histograms of GT
+                    # if iCycle .. first middle last
+                    # save hist GT
+                
+                # \todo: Save meanGT at end of cycles.
+                            
+            # Sample for next cycle
+            sample = np.random.random(SAMPLE_COUNT)*MAXIMUM_NR_AGENTS
+            simEnv[0:SAMPLE_COUNT] = simEnv[sample]
```

Comment:
This and previous line I don't understand. `np.random.random(size)` gives a vector of floats and you use these to slice the `simEnv` array. 

First, floats should not be used to index arrays.
Second, why do the objects in `simEnv` need moving and why in random fashion.

Finallly you can omit `0` in slicing arrays by `simEnv[:SAMPLE_COUNT]`.